api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -206,6 +205,57 @@ message ExternalProcessor {
   // Instead, the stream to the external processor will be closed. There will be no
   // more external processing for this stream from now on.
   bool disable_immediate_response = 15;
+
+  // Options related to the sending and receiving of dynamic metadata.
+  MetadataOptions metadata_options = 16;
+}
+
+// The MetadataOptions structure defines options for the sending and receiving of
+// dynamic metadata. Specifically, which namespaces to send to the server, whether
+// metadata returned by the server may be written, and how that metadata may be written.
+message MetadataOptions {
+  // Defines how overrides apply when evaluating more-specific config against
+  // less-specific config. Defaults to "merge".
+  enum MergeBehavior {
+    // Merge the more-specific config's namespace list into the less-specific config's
+    // namespace list. With this option, more-specific config is always a strict superset
+    // of less-specific config.
+    merge = 0;
```

Comment:
These are used in the `FilterConfigPerRoute` constructors in ext_proc.cc. I will incorporate the case change and in the process I will use the API enum directly as @tyxia mentioned instead of defining an intermediate enum. 

This field is designed to define how more-specific config interacts with less-specific config, scoped to `MetadataOptions`'s namespaces fields. `MetadataOptions` is used in both `ExternalProcessor` to configure the filter and `ExtProcPerRoute` as an optional override.

Reply:
I might be missing it, but I don't see it defined anywhere what happens when there are conflicts between less specific and more specific.  For example, if the less specific config defines metadata `A`, and the more specific defines metadata `B`, I assume merge behavior would mean the final metadata consists of both `A` and `B`.  However, what happens if both less-specific and more-specific define `A`?

Reply:
Alternatively, why not require the more specific thing to specify all the metadata it requires from the less specific thing?  Then you can always overwrite more specific thing over less specific thing.